### PR TITLE
add perovskite db

### LINKED ISSUE #419: add perovskite db

### PR DESCRIPTION
**Pull request by kjappelbaum**
Fixes #419

### CODE REVIEW

**Comment by kjappelbaum** [comment]
dropped a lot of data to keep the setup simple for now

File: data/tabular/perovskite_db/perovskite_solar_cells_with_descriptions.csv
Comment (MicPie) [inline data/tabular/perovskite_db/perovskite_solar_cells_with_descriptions.csv:1]: Is this file needed in the dir?

Comment (MicPie) [reply data/tabular/perovskite_db/perovskite_solar_cells_with_descriptions.csv:1]: I.e., can't we download it from the source?

File: data/tabular/perovskite_db/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
+name: perovskite_db
+description: |-
+  Database of perovskite solar cells with their composition, device stacks, and performance.
+targets:
```
Comment (MicPie) [inline data/tabular/perovskite_db/meta.yaml:5]: Below you round to two digits. If this is the desired precision we should set it here with `significant_digits` otherwise it will get overridden by current the default = 3.

**Review by MicPie — CHANGES_REQUESTED** [review]
Please check if we can download it directly and if the precision needs to be set. :-)

File: data/tabular/perovskite_db/perovskite_solar_cells_with_descriptions.csv
Comment (kjappelbaum) [reply data/tabular/perovskite_db/perovskite_solar_cells_with_descriptions.csv:1]: I can upload it to HF or somewhere else.

File: data/tabular/perovskite_db/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        title = {An open-access database and analysis tool for perovskite solar cells based on the {FAIR} data principles},
+        journal = {Nat Energy}
+      }
+templates:
+    - The perovskite material with the {reduced_formulas__description} of {reduced_formulas#} has a bandgap of {bandgap#} eV.
```
Comment (MicPie) [inline data/tabular/perovskite_db/meta.yaml:76]: instead of `eV` you could use `{TARGET__units}`

Comment (kjappelbaum) [reply data/tabular/perovskite_db/meta.yaml:76]: but is there currently an advantage in doing this?

File: data/tabular/perovskite_db/meta.yaml (hunk 1 above)
Comment (MicPie) [reply data/tabular/perovskite_db/meta.yaml:5]: It should, here is the code: https://github.com/OpenBioML/chemnlp/blob/main/data/text_sampling/text_sampling.py#L539

File: data/tabular/perovskite_db/meta.yaml (hunk 2 above)
Comment (MicPie) [reply data/tabular/perovskite_db/meta.yaml:76]: Not if you won't change this dataset setup anymore or if you won't reuse the templates. ;-)

File: data/tabular/perovskite_db/meta.yaml (hunk 1 above)
Comment (kjappelbaum) [reply data/tabular/perovskite_db/meta.yaml:5]: i'll try to reproduce and open an issue in this case, but I am quite sure I got a lot of digits

File: data/tabular/perovskite_db/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,155 @@
+---
+name: perovskite_db
+description: |-
+  Database of perovskite solar cells with their composition, device stacks, and performance.
+targets:
+    - id: bandgap
+      description: bandgap of the perovskite material
+      units: eV
+      type: float
```
Comment (MicPie) [inline data/tabular/perovskite_db/meta.yaml:9]: I guess you need to change `type: float` to `type: continous`?

Comment (MicPie) [reply data/tabular/perovskite_db/meta.yaml:9]: yep: https://github.com/OpenBioML/chemnlp/blob/main/src/chemnlp/data_val/model.py#L35

Comment (kjappelbaum) [reply data/tabular/perovskite_db/meta.yaml]: addressed

File: data/tabular/perovskite_db/meta.yaml (hunk 2 above)
Comment (kjappelbaum) [reply data/tabular/perovskite_db/meta.yaml]: updated.

File: data/tabular/perovskite_db/meta.yaml (hunk 1 above)
Comment (kjappelbaum) [reply data/tabular/perovskite_db/meta.yaml:5]: updated.

File: data/tabular/perovskite_db/meta.yaml [hunk 4]
```diff
@@ -1,49 +1,54 @@
...
 name: perovskite_db
 description: |-
-  Database of perovskite solar cells with their composition, device stacks, and performance.
+    Database of perovskite solar cells with their composition, device stacks, and performance.
 targets:
     - id: bandgap
       description: bandgap of the perovskite material
       units: eV
-      type: float
+      type: continuos
```
Comment (MicPie) [inline data/tabular/perovskite_db/meta.yaml:9]: typo: `continuos`--> `continuous`

**Review by MicPie — CHANGES_REQUESTED** [review]
Changes lgtm, only one small typo that needs a fix. 👍

**Review by kjappelbaum — COMMENTED** [review]
typo

**Review by MicPie — APPROVED** [review]
lgtm
